annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastFieldResolver.java

```diff
@@ -6,37 +6,52 @@
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.type.TypeMirror;
-import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
 import java.util.*;
 import java.util.stream.Collectors;
 
 class FastFieldResolver {
     //Stores List of Fast fields for class elements
     private HashMap<Element, List<FastFieldElement>> cache = new HashMap<>();
 
-    private Elements elements;
+    private final Types typeUtils;
 
-    public FastFieldResolver(Elements elements) {
-        this.elements = elements;
+    public FastFieldResolver(Types typeUtils) {
+        this.typeUtils = typeUtils;
     }
 
-    private boolean isOverriddenIn(FastTypeElement fastType, FastFieldElement fastField) {
+    private boolean isValidAndOverriddenIn(FastTypeElement fastType, FastFieldElement fastField) {
         List<FastFieldElement> fastFieldElements = getFastFieldsFor(fastType.getElement());
-        return fastFieldElements.stream().anyMatch(field -> field.getMethodName().equals(fastField.getMethodName()) &&
-            field.getParameterType().equals(fastField.getParameterType()));
+        List<TypeMirror> fastFieldMethodParameters = getMethodParametersTypeMirrors(fastField);
+        return fastFieldElements.stream().anyMatch(fastFieldElement -> {
+            List<TypeMirror> parameters = getMethodParametersTypeMirrors(fastFieldElement);
+            return fastFieldElement.getMethodName().equals(fastField.getMethodName()) &&
+                fastFieldMethodParameters.size() == parameters.size() &&
+                fastFieldMethodParameters.equals(parameters);
+        });
+    }
+
+    private List<TypeMirror> getMethodParametersTypeMirrors(FastFieldElement fastField) {
+        return fastField.getFastField().getParameters()
+            .stream().map(Element::asType).collect(Collectors.toList());
     }
 
     private List<FastFieldElement> getFastFieldsFor(Element aClass) {
+
         return cache.computeIfAbsent(aClass,
             key -> key.getEnclosedElements().stream()
-            .filter(element -> element.getAnnotation(FastField.class) != null)
-            .map(element -> new FastFieldElement((ExecutableElement) element))
-            .collect(Collectors.toList()));
+                .filter(element -> element.getAnnotation(FastField.class) != null)
+                .map(element -> new FastFieldElement((ExecutableElement) element))
+                .collect(Collectors.toList()));
     }
 
     private Element getSuperClass(Element element) {
         TypeMirror superClass = ((TypeElement) element).getSuperclass();
-        return elements.getTypeElement(superClass.toString());
+        return typeUtils.asElement(superClass);
+    }
+
+    private boolean containsInvalidMethodParameters(List<FastFieldElement> fastFieldElements) {
+        return fastFieldElements.stream().anyMatch(field -> field.getParametersNumber() != 1);
     }
 
     public void resolveFields(FastTypeElement fastType) {
```

Comment:
I think in the following scenario: 

```java
@FastType
class A extends B {
}

class B extends C {
@FastField
void setFoo(int value) { }
}

class C {
@FastField
void setFoo(int value) { }
}
```
we'll get duplicate field  'Foo' error while shouldn't. Please check